src/main/java/com/linkedin/kmf/services/ConsumerFactory.java

```diff
@@ -0,0 +1,108 @@
+/**
+ * Copyright 2016 LinkedIn Corp. Licensed under the Apache License, Version 2.0 (the "License"); you may not use this
+ * file except in compliance with the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package com.linkedin.kmf.services;
+
+import com.linkedin.kmf.consumer.KMBaseConsumer;
+import com.linkedin.kmf.consumer.NewConsumer;
+import com.linkedin.kmf.services.configs.CommonServiceConfig;
+import com.linkedin.kmf.services.configs.ConsumeServiceConfig;
+import java.lang.reflect.InvocationTargetException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Random;
+import org.apache.kafka.clients.admin.AdminClient;
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.common.config.ConfigException;
+import org.apache.kafka.common.serialization.StringDeserializer;
+
+
+public class ConsumerFactory {
```

Comment:
If ConsumerFactory is an interface then it will be a little easier to mock.  This implementation then moves to ConsumerFactoryImpl (or whatever you want to call it).